### PR TITLE
[cisco_secure_email_gateway] Support New Format for Mail Category of logs and Add Pipeline for Antivirus

### PR DESCRIPTION
**Pull request by mohitjha-elastic**
Type of change
- Enhancement

## What does this PR do?
1. **Support new format for mail category of logs.**
Add grok patterns to parse new logs for the text_mail category.
Below are the new sample logs for which we add the category:
```
<166>Sep 12 11:00:00 mail_logs: Info: Message finished MID 68119155 done
<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 SDR: Consolidated Sender Threat Level: Neutral, Threat Category: N/A, Suspected Domain(s) : N/A (other reasons for verdict). Sender Maturity: 30 days (or greater) for domain: redacted.redactedMailFrom.com
<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 DKIM: pass signature verified (d=redactedMailFrom.com s=srsa2048 i=@redactedMailFrom.com)
<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 DMARC: Message from domain redactedMailFrom.com, DMARC pass (SPF aligned True, DKIM aligned True)
<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 SPF: mailfrom identity no-reply@redactedMailFrom.com Pass (v=spf1)
```

For more examples of logs, check the issue [here](https://github.com/elastic/integrations/issues/6736).

2.  **Add pipeline for antivirus category.**
Add ingest pipeline for the antivirus category to extract the value of fields.
Sample logs that we support are attached below:
```
<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 69391938 - Result 'CLEAN' ()
<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 68431780 0 - Error - 'Encrypted' '0x8004021'
<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66784457 0 - Virus 'CXmail/MalPE-HB' 'body.scan/242426.cab/rockro9046.exe' 1 0
<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66710307 7 - Limit - 'Max Files Exceeded'
<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66708787 - timed out on message
```

For more examples of logs, check the issue [here](https://github.com/elastic/integrations/issues/6737).
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.7.1

## How to test this PR locally
Clone integrations repo.
Install the elastic package locally.
Start the elastic stack using the elastic package.
Move to integrations/packages/cisco_secure_email_gateway directory.
Run the following command to run tests.
`elastic-package test -v`

## Related issues

- Closes https://github.com/elastic/integrations/issues/6737
- Closes https://github.com/elastic/integrations/issues/6736

## Automated Test

[test-cisco_secure_email_gateway.txt](https://github.com/elastic/integrations/files/13166929/test-cisco_secure_email_gateway.txt)

### CODE REVIEW

File: packages/cisco_secure_email_gateway/_dev/deploy/docker/sample_logs/antivirus.@123456789.s [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 69391938 - Result 'CLEAN' ()
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 68431780 0 - Error - 'Encrypted' '0x8004021'
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 66842418 0 - Virus 'CXmail/Phish-O' 'body.scan/Payment.html' 1 0
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 68325092 0 - Virus 'CXmail/MalPE-AC' 'body.scan/redactedFileName.rar' 1 0
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 66301278 0 - Virus 'CXmail/MalPE-HB' 'body.scan/anotherFileName.arj/anotherFileName.exe' 1 0
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 66298546 0 - Virus 'Mal/DrodRar-AIC' 'body.scan/anotherFileName.arj' 1 0
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 67753636 0 - Virus 'Troj/MSIL-TAR' 'body.scan/otherFileName.exe' 1 0
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 66710307 7 - Limit - 'Max Files Exceeded'
+Fri Sep 12 11:00:00 2023 Info: sophos  antivirus - MID 66708787 - timed out on message
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/_dev/deploy/docker/sample_logs/antivirus.@123456789.s:9]: Please add a new line here, this line is not currently tested as logfile only sends newline-terminated lines.

File: packages/cisco_secure_email_gateway/_dev/deploy/docker/sample_logs/mail_logs.@123456789.s [hunk 2]
```diff
@@ -18,3 +18,22 @@ Fri Mar 25 20:50:10 2022 Warning: Your "IronPort Anti-Spam" key will expire in u
...
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 matched all recipients for per-recipient policy DEFAULT in the inbound table
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 SDR: Tracker Header : redactedTrackerHeader
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 SDR: Domains for which SDR is requested: reverse DNS host: redacted.redactedMailFrom.com, helo: redacted.redactedMailFrom.com, env-from: redactedMailFrom.com, header-from: redactedMailFrom.com, reply-to: redactedMailFrom.com
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 SDR: Consolidated Sender Threat Level: Neutral, Threat Category: N/A, Suspected Domain(s) : N/A (other reasons for verdict). Sender Maturity: 30 days (or greater) for domain: redacted.redactedMailFrom.com
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 DMARC: Message from domain redactedMailFrom.com, DMARC pass (SPF aligned True, DKIM aligned True)
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 DKIM: pass signature verified (d=redactedMailFrom.com s=srsa2048 i=@redactedMailFrom.com)
+Fri Sep 12 11:00:00 2022 Info: MID 68119155 AMP file reputation verdict : SKIPPED (no attachment in message)
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/_dev/deploy/docker/sample_logs/mail_logs.@123456789.s:39]: Please add final new line.

File: packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-antivirus.log [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 68431780 0 - Error - 'Encrypted' '0x8004021'
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66842418 0 - Virus 'CXmail/Phish-O' 'body.scan/Payment.html' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66784457 0 - Virus 'CXmail/MalPE-HB' 'body.scan/242426.cab/rockro9046.exe' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 68016096 0 - Virus 'CXmail/MalPE-FL' 'body.scan/redactedFileName.rar/redactedFileName.exe' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 68016096 0 - Virus 'CXmail/MalPE-AC' 'body.scan/redactedFileName.rar' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66301278 0 - Virus 'Mal/DrodRar-AIC' 'body.scan/anotherFileName.arj' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 67753636 0 - Virus 'Troj/MSIL-TAR' 'body.scan/otherFileName.exe' 1 0
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66710307 7 - Limit - 'Max Files Exceeded'
+<166>Sep 12 11:00:00 antivirus: Info: sophos  antivirus - MID 66708787 - timed out on message
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-antivirus.log:10]: New line is not essential here, but please add anyway.

File: packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log [hunk 4]
```diff
@@ -18,3 +18,22 @@
...
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 matched all recipients for per-recipient policy DEFAULT in the inbound table
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 SDR: Tracker Header : redactedTrackerHeader
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 SDR: Domains for which SDR is requested: reverse DNS host: redacted.redactedMailFrom.com, helo: redacted.redactedMailFrom.com, env-from: redactedMailFrom.com, header-from: redactedMailFrom.com, reply-to: redactedMailFrom.com
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 SDR: Consolidated Sender Threat Level: Neutral, Threat Category: N/A, Suspected Domain(s) : N/A (other reasons for verdict). Sender Maturity: 30 days (or greater) for domain: redacted.redactedMailFrom.com
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 DMARC: Message from domain redactedMailFrom.com, DMARC pass (SPF aligned True, DKIM aligned True)
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 DKIM: pass signature verified (d=redactedMailFrom.com s=srsa2048 i=@redactedMailFrom.com)
+<166>Sep 12 11:00:00 mail_logs: Info: MID 68119155 AMP file reputation verdict : SKIPPED (no attachment in message)
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log:39]: Same here; not required to add, but please do.

File: packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log-expected.json [hunk 5]
```diff
@@ -716,6 +716,665 @@
...
+        },
+        {
+            "@timestamp": "2023-09-12T11:00:00.000Z",
+            "cisco_secure_email_gateway": {
+                "log": {
+                    "category": {
+                        "name": "mail_logs"
+                    },
+                    "message": "MID 68119155 Subject \"redacted subject\"",
+                    "subject": "\"redacted subject\""
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log-expected.json:761]: ISTM that the subject should be unquoted.

File: packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log-expected.json [hunk 6]
```diff
@@ -716,6 +716,665 @@
...
+            ]
+        },
+        {
+            "@timestamp": "2023-09-12T11:00:00.000Z",
+            "cisco_secure_email_gateway": {
+                "log": {
+                    "category": {
+                        "name": "mail_logs"
+                    },
+                    "dkim_aligned": "True",
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-text-mail.log-expected.json:1273]: Can this be converted to boolean? Similar for `spf_aligned`.

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+processors:
+  - set:
+      field: event.kind
+      value: event
+  - grok:
+      field: cisco_secure_email_gateway.log.message
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml:7]: Please add a tag.

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+      field: event.type
+      if: ctx.cisco_secure_email_gateway?.log?.type == 'Error'
+      value: error
+on_failure:
+  - set:
+      field: event.kind
+      value: pipeline_error
+  - append:
+      field: error.message
+      value: '{{{ _ingest.on_failure_message }}}'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml:25]:
```suggestion
  - append:
      field: error.message
      value: |-
        Processor "{{{ _ingest.on_failure_processor_type }}}" with tag "{{{ _ingest.on_failure_processor_tag }}}" in pipeline "{{{ _ingest.on_failure_pipeline }}}" failed with message "{{{ _ingest.on_failure_message }}}"
```

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 9]
```diff
@@ -20,11 +20,29 @@ processors:
...
         - "^DCID %{NUMBER:cisco_secure_email_gateway.log.delivery_connection_id} %{WORD:cisco_secure_email_gateway.log.connection_status}$"
-        - "^%{GREEDYDATA:cisco_secure_email_gateway.log.subject}\\. %{GREEDYDATA:cisco_secure_email_gateway.log.description}$"
         - "^%{GREEDYDATA:cisco_secure_email_gateway.log.description}: Unable to send System/Warning %{DATA:event.kind} to %{DATA:email.to.address} with subject \"%{GREEDYDATA:email.subject}\".$"
         - "^Internal %{DATA:network.protocol} system successfully sent a message to %{DATA:email.to.address} with subject %{GREEDYDATA:email.subject}.$"
         - "^Internal %{DATA:network.protocol} giving up on message to %{DATA:email.to.address} with subject %{GREEDYDATA:email.subject}.$"
         - "^Internal %{DATA:network.protocol} Error: %{GREEDYDATA:cisco_secure_email_gateway.log.description} to host %{IP:destination.ip}:%{NUMBER:destination.port:long} for recipient %{DATA:email.to.address}: %{GREEDYDATA:cisco_secure_email_gateway.log.subject}$"
+        - "^MID %{NUMBER:email.message_id} RID \\[%{DATA:cisco_secure_email_gateway.log.recipient_id}\\] Response %{GREEDYDATA:cisco_secure_email_gateway.log.response}$"
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:27]: Please use single quotes to reduce escape proliferation. (may as well fix up the patterns above as well)

File: packages/cisco_secure_email_gateway/data_stream/log/fields/fields.yml
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/fields/fields.yml:1]: Can we have descriptions for fields?

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        - "^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} - \\'%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}\\' \\'%{GREEDYDATA:cisco_secure_email_gateway.log.encrypted_hash}\\'$"
+        - "^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} \\'%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}\\' \\'body.scan\\/%{GREEDYDATA:file.name}\\' 1 0$"
+        - "^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} - \\'%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}\\'$"
+        - "^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} - %{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}$"
+        - "^%{GREEDYDATA:cisco_secure_email_gateway.log.message}$"
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_antivirus.yml:14]:
```suggestion
        - '^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} - %{WORD:cisco_secure_email_gateway.log.type} ''%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}'' \(\)$'
        - '^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} - ''%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}'' ''%{GREEDYDATA:cisco_secure_email_gateway.log.encrypted_hash}''$'
        - '^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} ''%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}'' ''body.scan\/%{GREEDYDATA:file.name}'' 1 0$'
        - '^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} %{NUMBER:cisco_secure_email_gateway.log.rank:long} - %{WORD:cisco_secure_email_gateway.log.type} - ''%{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}''$'
        - '^%{WORD:observer.vendor}  antivirus - MID %{NUMBER:email.message_id} - %{GREEDYDATA:cisco_secure_email_gateway.log.antivirus_result}$'
        - '^%{GREEDYDATA:cisco_secure_email_gateway.log.message}$'
```

File: packages/cisco_secure_email_gateway/data_stream/log/fields/fields.yml
Comment (mohitjha-elastic) [reply packages/cisco_secure_email_gateway/data_stream/log/fields/fields.yml:1]: We have only added descriptions for the fields that are available.

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 11]
```diff
@@ -6,33 +6,89 @@ processors:
...
+        - '^DCID %{NUMBER:cisco_secure_email_gateway.log.delivery_connection_id} %{WORD:cisco_secure_email_gateway.log.connection_status}$'
+        - '^%{GREEDYDATA:cisco_secure_email_gateway.log.description}: Unable to send System/Warning %{DATA:event.kind} to %{DATA:email.to.address} with subject \"%{GREEDYDATA:email.subject}\".$'
+        - '^Internal %{DATA:network.protocol} system successfully sent a message to %{DATA:email.to.address} with subject %{GREEDYDATA:email.subject}.$'
+        - '^Internal %{DATA:network.protocol} giving up on message to %{DATA:email.to.address} with subject %{GREEDYDATA:email.subject}.$'
+        - '^Internal %{DATA:network.protocol} Error: %{GREEDYDATA:cisco_secure_email_gateway.log.description} to host %{IP:destination.ip}:%{NUMBER:destination.port:long} for recipient %{DATA:email.to.address}: %{GREEDYDATA:cisco_secure_email_gateway.log.subject}$'
+        - '^MID %{NUMBER:email.message_id} RID \[%{DATA:cisco_secure_email_gateway.log.recipient_id}\] Response %{GREEDYDATA:cisco_secure_email_gateway.log.response}$'
+        - '^MID %{NUMBER:email.message_id} Subject \"%{GREEDYDATA:cisco_secure_email_gateway.log.subject}\"$'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:28]:
```suggestion
        - '^MID %{NUMBER:email.message_id} Subject "%{GREEDYDATA:cisco_secure_email_gateway.log.subject}"$'
```
?

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 12]
```diff
@@ -6,33 +6,89 @@ processors:
...
+        - '^MID %{NUMBER:email.message_id} %{GREEDYDATA:cisco_secure_email_gateway.log.message_status} for delivery$'
+        - '^Message %{GREEDYDATA:cisco_secure_email_gateway.log.message_status} MID %{NUMBER:email.message_id} done$'
+        - '^MID %{NUMBER:email.message_id} interim verdict using engine\: %{GREEDYDATA:cisco_secure_email_gateway.log.engine}$'
+        - '^MID %{NUMBER:email.message_id} interim AV verdict using %{GREEDYDATA:cisco_secure_email_gateway.log.engine}$'
+        - '^MID %{NUMBER:email.message_id} using engine\: %{GREEDYDATA:cisco_secure_email_gateway.log.engine}$'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:33]:
```suggestion
        - '^MID %{NUMBER:email.message_id} using engine: %{GREEDYDATA:cisco_secure_email_gateway.log.engine}$'
```
? (similar below)

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 13]
```diff
@@ -6,33 +6,89 @@ processors:
...
+        - '^MID %{NUMBER:email.message_id} %{GREEDYDATA:cisco_secure_email_gateway.log.subject}\: verdict %{GREEDYDATA:cisco_secure_email_gateway.log.verdict_scale}$'
+        - '^MID %{NUMBER:email.message_id} Message-ID ''\<%{GREEDYDATA:cisco_secure_email_gateway.log.email}\>\''$'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:35]:
```suggestion
        - '^MID %{NUMBER:email.message_id} Message-ID ''\<%{GREEDYDATA:cisco_secure_email_gateway.log.email}\>''$'
```
?

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 14]
```diff
@@ -6,33 +6,89 @@ processors:
...
+        - '^MID %{NUMBER:email.message_id} %{DATA:network.protocol}\: Verification %{GREEDYDATA:cisco_secure_email_gateway.log.verified}$'
+        - '^MID %{NUMBER:email.message_id} %{DATA:network.protocol}\: mailfrom identity %{DATA:email.from.address} %{GREEDYDATA:cisco_secure_email_gateway.log.verified} \(v\=spf1\)$'
+        - '^MID %{NUMBER:email.message_id} matched all recipients for per-recipient policy %{WORD:cisco_secure_email_gateway.log.policy} in the %{DATA:email.direction} table$'
+        - '^MID %{NUMBER:email.message_id} SDR\: Tracker Header \: %{GREEDYDATA:cisco_secure_email_gateway.log.email_tracker_header}$'
+        - '^MID %{NUMBER:email.message_id} SDR\: Domains for which SDR is requested\: reverse DNS host\: %{DATA:dns.question.name}\, helo\: %{DATA:cisco_secure_email_gateway.log.helo}\, env-from\: %{DATA:cisco_secure_email_gateway.log.env}\, header-from\: %{DATA:email.from.address}\, reply-to\: %{DATA:email.to.address}$'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:40]:
```suggestion
        - '^MID %{NUMBER:email.message_id} SDR: Domains for which SDR is requested: reverse DNS host: %{DATA:dns.question.name}, helo: %{DATA:cisco_secure_email_gateway.log.helo}, env-from: %{DATA:cisco_secure_email_gateway.log.env}, header-from: %{DATA:email.from.address}, reply-to: %{DATA:email.to.address}$'
```
? (similar elsewhere)

File: packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml [hunk 15]
```diff
@@ -6,33 +6,89 @@ processors:
...
+        - '^%{GREEDYDATA:cisco_secure_email_gateway.log.message_status} DCID %{NUMBER:cisco_secure_email_gateway.log.delivery_connection_id} MID %{NUMBER:email.message_id} to RID \[%{DATA:cisco_secure_email_gateway.log.recipient_id}\]$'
+        - '^%{GREEDYDATA:cisco_secure_email_gateway.log.message_status} DCID %{NUMBER:cisco_secure_email_gateway.log.delivery_connection_id} MID %{NUMBER:email.message_id} to RID \[%{DATA:cisco_secure_email_gateway.log.recipient_id}\]$'
+        - '^DCID %{NUMBER:cisco_secure_email_gateway.log.delivery_connection_id} %{WORD:cisco_secure_email_gateway.log.connection_status}$'
+        - '^%{GREEDYDATA:cisco_secure_email_gateway.log.description}: Unable to send System/Warning %{DATA:event.kind} to %{DATA:email.to.address} with subject "%{GREEDYDATA:email.subject}".$'
+        - '^Internal %{DATA:network.protocol} system successfully sent a message to %{DATA:email.to.address} with subject %{GREEDYDATA:email.subject}.$'
```
Comment (efd6) [inline packages/cisco_secure_email_gateway/data_stream/log/elasticsearch/ingest_pipeline/pipeline_text_mail_logs.yml:24]: I suspect the final `.` should be `\.`. I think there are other examples of this elsewhere in other files (as well as below).

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**